### ISSUE TITLE
tests/kernel/mem_protect/mem_map/kernel.memory_protection.mem_map failing on qemu_cortex_a53/qemu_cortex_a53/smp

### ISSUE THREAD

**Issue by aescolar** [zephyrproject-rtos/zephyr#74247]
**Describe the bug**
tests/kernel/mem_protect/mem_map/kernel.memory_protection.mem_map 
is failing on qemu_cortex_a53/qemu_cortex_a53/smp

**To Reproduce**
1. twister -p qemu_cortex_a53/qemu_cortex_a53/smp -s tests/kernel/mem_protect/mem_map/kernel.memory_protection.mem_map -T tests/kernel/mem_protect/mem_map 

**Expected behavior**
No errors

**Impact**
main CI failing

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/actions/runs/9497518288/job/26174375295
https://github.com/zephyrproject-rtos/zephyr/actions/runs/9497518288/job/26174375523

```
*** Booting Zephyr OS build v3.6.0-6124-gf7e11649fdf5 ***
Secondary CPU core 1 (MPID:0x1) is up
Running TESTSUITE mem_map
....
Running TESTSUITE mem_map_api
===================================================================
START - test_k_mem_map_exhaustion
Free memory: 133599232
E: insufficient virtual address space (requested 12288)
Mapped 628 pages
Free memory now: 131026944
ASSERTION FAIL [is_table_desc(src_table[i], level)] @ ZEPHYR_BASE/arch/arm64/core/mmu.c:486
        can't have partial block pte here
E: ELR_ELn: 0x0000000040002f70
E: ESR_ELn: 0x0000000056000002
E:   EC:  0x15 (Unknown)
E:   IL:  0x1
E:   ISS: 0x2
E: TPIDRRO: 0x0100000040045dd8
E: x0:  0x0000000040045dd8  x1:  0x00000000000001e6
E: x2:  0x0000000000000000  x3:  0xffffff80ffffffc8
E: x4:  0x0000000000000000  x5:  0x000000000000002d
E: x6:  0x0000000000000000  x7:  0x0000000000000004
E: x8:  0x0000000000000004  x9:  0x0000000040006ed0
E: x10: 0x000000004001aef8  x11: 0x0000000000000000
E: x12: 0x0000000000000000  x13: 0x000000009600004f
E: x14: 0x0000000040092f10  x15: 0x00000000000003c0
E: x16: 0x0000000040008af8  x17: 0x00000000400455e0
E: x18: 0x0000000040044b00  lr:  0x00000000400059cc
E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
E: Current thread: 0x40044b00 (test_k_mem_map_exhaustion)
Caught system error -- reason 4
Unexpected fault during test
===================================================================
RunID: 43f2e968041ce2e6e08754a4ac2874be
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04
 - Toolchain (e.g Zephyr SDK, ...): 0.16.8
 - Commit SHA or Version used: main after f7e11649fdf578b2e958e460c96566216dd57ff1

**Additional context**
Introduced by f7e11649fdf578b2e958e460c96566216dd57ff1

**Comment by carlocaione** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/72706#issuecomment-2165333454

**Comment by npitre** [comment]
Fix in PR #74269
